### PR TITLE
Feature/infisical support

### PR DESCRIPTION
**Pull request by Nonobis**
Added support for infisical along bitwarden client.

-----

I have attempted to add support for infisical but how to test ??
My implementation seem to be ok (at leat the connection)

Loading collectors from /usr/app/src/collectors
11 collectors loaded: amazon, bureau_vallee, carrefour, free, intermarche, leclerc, leroy_merlin, orange, ovh, total_energies, u    
Collection Task started! Invoice collection will be performed every hour.
App listening on port 8080
Connected successfully to MongoDB
No customer found in database, creating default customer.
Default customer created. Bearer is "XXXXX". Keep it safe, it will not be displayed again.       
Connected successfully to iNFISICAL

My infisical client seem to be ok, but on locvalhost:8080  seeem to be only api.
I think i need the registry ? or i can only test via api ?

Thnkas

### CODE REVIEW

**Comment by LucBerge** [comment]
Hello @Nonobis,
I just saw the PR right now 😅 so sorry for the delay.

> I have attempted to add support for infisical but how to test ??

In order to test, you can start the docker-container with `docker-compose -f docker-compose-debug-infisical.yml up --build`. I will let you test first before merging the PR in a new branch called `infisical`. From there, I will test it again, update the doc and merge on master.

Can you please pull the latest version of `master` on your repo?
I have added work that will simplify the code.

File: docker-compose-debug-infisical.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    depends_on:
+      - mongodb
+    environment:
+      - PORT=8080
+      - REGISTRY_SERVER_ENDPOINT=https://registry.invoice-collector.com
+      - DATABASE_URI=mongodb://mongodb:27017
+      - DATABASE_MONGODB_NAME=dev
+      - SECRET_MANAGER_TYPE=infisical # bitwarden or infisical
+      - SECRET_MANAGER_BITWARDEN_API_URI=https://vault.bitwarden.eu/api
+      - SECRET_MANAGER_BITWARDEN_IDENTITY_URI=https://vault.bitwarden.eu/identity
```
Comment (LucBerge) [inline docker-compose-debug-infisical.yml:25]: Remove all bitwarden env vars

File: src/secret_manager/infisical.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        }
+        if (!process.env.SECRET_MANAGER_INFISICAL_PROJECT_ID) {
+            throw new Error("SECRET_MANAGER_INFISICAL_PROJECT_ID environment variable is required");
+        }
+        if (!process.env.SECRET_MANAGER_INFISICAL_ENV) {
+            throw new Error("SECRET_MANAGER_INFISICAL_ENV environment variable is required");
+        }
+
+        super();
+        this.clientId = process.env.SECRET_MANAGER_INFISICAL_CLIENT_ID || "";
```
Comment (LucBerge) [inline src/secret_manager/infisical.ts:30]: Merge master in your branch, then use `utils.getEnvVar`.
See https://github.com/invoice-collector/invoice-collector/blob/208ba8665430c5469b43361a2291e787f5b0865c/src/secret_manager/bitwarden.ts#L16C28-L16C43

Remove the if conditions in constructor as well.

File: src/secret_manager/infisical.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    async connect(): Promise<void> {
+        try {
+            // Authenticate with Infisical
+            await this.client.auth().universalAuth.login({
+                clientId: this.clientId,
+                clientSecret: this.clientSecret
+            });
+            console.log("Connected successfully to iNFISICAL");
```
Comment (LucBerge) [inline src/secret_manager/infisical.ts:47]: Infisical

File: src/secret_manager/infisical.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        }
+        catch (err) {
+            console.error("Connection to Bitwarden failed", err);
```
Comment (LucBerge) [inline src/secret_manager/infisical.ts:50]: Infisical

File: src/secret_manager/infisical.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            const secret = await this.client.secrets().getSecret({
+                projectId: this.projectId,
+                secretName: id,
+                environment: this.env,
+                secretPath: "/",
+              });
+            return JSON.parse(secret.secretValue);
+        }
+        catch (err) {
+            if (err instanceof Error && err.message.includes("[404 Not Found]")) {
```
Comment (LucBerge) [inline src/secret_manager/infisical.ts:79]: In Bitwarden implementation, this error handling is for when the credential is in the mongodb but not in the secret manager.

Error handling here might be different from Bitwarden.

File: src/secret_manager/infisical.ts [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+                    secretPath: "/"
+                  });
+                  console.log(`Secret "${secretName}" deleted.`);
+                } catch (error) {
+                  console.error(`Error deleting "${secretName}":`, error);
+                }
+              }
+        }
+        catch (err) {
+            if (!(err instanceof Error && err.message.includes("[404 Not Found]"))) {
```
Comment (LucBerge) [inline src/secret_manager/infisical.ts:106]: In Bitwarden implementation, this error handling is for when the credential is in the mongodb but not in the secret manager.

Error handling here might be different from Bitwarden.

File: src/secret_manager/secretManagerFactory.ts [hunk 7]
```diff
@@ -10,6 +11,9 @@ export class SecretManagerFactory {
             switch(type) {
                 case 'bitwarden':
                     SecretManagerFactory.instance = new Bitwarden();
+                break;
```
Comment (LucBerge) [inline src/secret_manager/secretManagerFactory.ts:14]: Missing tab

File: docker-compose-debug-infisical.yml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+      dockerfile: Dockerfile
+    env_file:
+      - .env
+    ports:
+      - 8080:8080
+      - 9229:9229
+    volumes:
+      - ./src:/usr/app/src
+      - ./locales:/usr/app/locales
+      - ./views:/usr/app/views
```
Comment (LucBerge) [inline docker-compose-debug-infisical.yml:15]: Add
```
      - ./media:/usr/app/media
      - ./test:/usr/app/test
```

Comment (LucBerge) [reply docker-compose-debug-infisical.yml:15]: Will automatically appear when merging master in your branch

**Comment by Nonobis** [comment]
Hello,
I will try to update this weekend and fix your reviews.
I am used to docker, but i have a infisical instance on my proxmox cluster ... so forgot to add a local image :)

**Comment by Nonobis** [comment]
sorry just a few commit done this night ... i need to test from zero when i got time

**Comment by Nonobis** [comment]
image
